pkg/gatewayserver/gatewayserver.go

```diff
@@ -206,6 +214,24 @@ func New(c *component.Component, conf *Config, opts ...Option) (gs *GatewayServe
 
 	hooks.RegisterUnaryHook("/ttn.lorawan.v3.NsGs", cluster.HookName, c.ClusterAuthUnaryHook())
 
+	for name, prefix := range gs.forward {
+		str := strings.Split(name, ":")
+		if len(str) != 2 {
+			continue
+		}
+		switch str[0] {
+		case "ttn.lorawan.v3.GsNs":
+			gs.upstreamHandlers[str[1]] = ns.NewHandler(ctx, str[1], c, prefix)
+		default:
+			return nil, errUpstreamType.WithAttributes("name", name)
+		}
+	}
+	for _, handler := range gs.upstreamHandlers {
+		if err := handler.Setup(); err != nil {
+			return nil, errStartUpstream.WithCause(err).WithAttributes("name", handler.GetName())
```

Comment:
Is it setup or start? Consistency is key for your future self and other developers

Reply:
`setup` was what we agreed on in the TTI draft PR.

Reply:
OK, so `errSetupUpstream`